services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -150,13 +166,118 @@ class PortHistorySpec extends HealthCheckSpecification {
+            Long timestampAfterStat = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterStat)) {
+                it.size() == 3
+                def antiflapStat = it[-1]
+                checkPortHistory(antiflapStat, isl.srcSwitch.dpId, isl.srcPort,
+                        PortHistoryEvent.ANTI_FLAP_PERIODIC_STATS)
+                count == antiflapStat.downCount
+                count == antiflapStat.upCount
+            }
+        }
+
+        and: "Cleanup: revert system to original state"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval + antiflapCooldown) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+    }
+
+    def "System shows antiflap statistic in the ANTI_FLAP_DEACTIVATED event when antiflap is deactivated\
+ before collecting ANTI_FLAP_PERIODIC_STATS"() {
+        assumeTrue("It can't be run when antiflap.cooldown + flap_duration > port.antiflap.stats.dumping.interval.seconds",
+                antiflapCooldown + 3 < antiflapDumpingInterval)
+        //port up/down procedure is done once in this test, so it can't take more than 3 seconds
+
+        given: "A direct link"
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+        assumeTrue("Unable to find ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeMillis()
+
+        when: "Execute port DOWN on the src switch for activating antiflap"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                it.size() == 2
+            }
+        }
+
+        and: "Generate antiflap statistic"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Antiflap statistic is available in port history inside the ANTI_FLAP_DEACTIVATED event"
+        Wrappers.wait(antiflapCooldown + WAIT_OFFSET) {
+            Long timestampAfterStat = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterStat)) {
+                it.findAll { it.event == PortHistoryEvent.ANTI_FLAP_PERIODIC_STATS.toString() }.empty
+                it.size() == 3
+                def antiflapStat = it[-1]
+                checkPortHistory(antiflapStat, isl.srcSwitch.dpId, isl.srcPort,
+                        PortHistoryEvent.ANTI_FLAP_DEACTIVATED)
+                antiflapStat.downCount == 1
+                antiflapStat.upCount == 1
```

Comment:
Should we check that these values equal to `0` for no-antiflap tests?

Reply:
`"System doesn't return antiflap statistic in port history if no flapping occurs during cooldown"()` - removed.

needed verification's were added to the first test